### PR TITLE
ENH: reference workflow N4 improvements

### PR DESCRIPTION
**Pull request by eilidhmacnicol**
## Changes proposed in this pull request
This makes the rodent-specific arguments (made available by nipreps/niworkflows#630) available to the reference workflow for more robust N4 correction. This should make the reference image easier to mask.

Also worth noting that in this scenario, brain extraction is run only once (on the structural image), and the registration between structural and functional is run without masking.

~Also updated `FLIRTRPT` to include `--basescale 1` argument, based on the conversation had over at nipreps/fmriprep#2624, to overcome problems in coregistration when testing locally.~
EDIT: this has been removed and will be added to a separate PR

### CODE REVIEW

File: fprodents/workflows/base.py [hunk 1]
```diff
@@ -327,10 +325,6 @@ def init_single_subject_wf(subject_id):
               ('outputnode.template', 'inputnode.template'),
               ('outputnode.anat2std_xfm', 'inputnode.anat2std_xfm'),
               ('outputnode.std2anat_xfm', 'inputnode.std2anat_xfm')]),
-            (bold_ref_wf, brain_extraction_wf, [
-                ('outputnode.epi_ref_file', 'inputnode.in_files')]),
-            (brain_extraction_wf, func_preproc_wf,
-             [("outputnode.out_mask", "inputnode.bold_mask")]),
```
Comment (oesteban) [inline fprodents/workflows/base.py:333]: does `func_preproc_wf` not need an input mask anymore?

File: fprodents/workflows/bold/base.py [hunk 2]
```diff
@@ -242,6 +242,7 @@ def init_func_preproc_wf(bold_file):
     outputnode = pe.Node(
         niu.IdentityInterface(
             fields=[
+                "bold_mask",
```
Comment (oesteban) [inline fprodents/workflows/bold/base.py:245]: it seems it needs the input

Comment (oesteban) [reply fprodents/workflows/bold/base.py:245]: actually, i think this is
```suggestion
                "anat_mask",
```

File: fprodents/workflows/bold/registration.py [hunk 3]
```diff
@@ -105,8 +105,13 @@ def init_bold_reg_wf(
...
     coreg = pe.Node(
-        FLIRTRPT(dof=bold2t1w_dof, generate_report=True, uses_qform=True), name="coreg"
-    )
+        FLIRTRPT(
+            dof=bold2t1w_dof,
+            generate_report=True,
+            uses_qform=True,
+            args="-basescale 1"
+        ),
+        name="coreg")
```
Comment (oesteban) [inline fprodents/workflows/bold/registration.py:114]: this should be on a separate PR :)

**Review by oesteban — APPROVED** [review]
Looks good overall

File: fprodents/workflows/base.py (hunk 1 above)
Comment (eilidhmacnicol) [reply fprodents/workflows/base.py:333]: No - in this case we transform the `anat_mask` from `anat_preproc_wf` after registering the `ref_file` in `func_preproc_wf`

File: fprodents/workflows/bold/base.py (hunk 2 above)
Comment (eilidhmacnicol) [reply fprodents/workflows/bold/base.py:245]: Specifically, we output the mask calculated from the structural in bold space. Do you think it's better to call it `bold_mask`, `anat_mask_bold`, or something else entirely?

Comment (oesteban) [reply fprodents/workflows/bold/base.py:245]: sorry, I did not realize this is the outputnode, scratch my comment
